src/encoded/static/components/filegallery.js

```diff
@@ -1783,6 +1773,241 @@ InclusionSelector.propTypes = {
+            return replicate === keyValue[0];
+        }
+        if (key === 'assembly') {
+            if (keyValue[0] === 'All assemblies') {
+                return true;
+            }
+            return (el[key] === keyValue[0]);
+        }
+        return el[key] === keyValue[0];
+    });
+}
+
+// Create objects for non-Assembly facets
+function createFacetObject(propertyKey, fileList, filters) {
+    // Initialize facet object
+    const facetObject = {};
+    // 'singleFilter' checks to see if there is only one filter selected (Assembly)
+    const singleFilter = Object.keys(filters).length === 1;
+    // Create list of files that satisfy all filters ('fileListFiltered')
+    // Create list of files that is filtered by assembly only ('newFileList')
+    let fileListFiltered = [...fileList];
+    let newFileList = [...fileList];
+    Object.keys(filters).forEach((filter) => {
+        if (filter === 'assembly') {
+            newFileList = filterItems(newFileList, filter, filters[filter]);
+        }
+        fileListFiltered = filterItems(fileListFiltered, filter, filters[filter]);
+    });
+    // If only 'Assembly' filter is selected, we want to collect all possible terms from 'newFile List' and then we are done
+    if (singleFilter) {
+        newFileList.forEach((file) => {
+            let property = file[propertyKey];
+            if (propertyKey === 'biological_replicates') {
+                property = (file.biological_replicates ? file.biological_replicates.sort((a, b) => a - b).join(', ') : '');
+            }
+            if (facetObject[property]) {
+                facetObject[property] += 1;
+            } else {
+                facetObject[property] = 1;
+            }
+        });
+    // If multiple filters are selected, it gets more complicated
+    } else {
+        // Start by adding properties from filtered list of files
+        fileListFiltered.forEach((file) => {
+            let property = file[propertyKey];
+            if (propertyKey === 'biological_replicates') {
+                property = (file.biological_replicates ? file.biological_replicates.sort((a, b) => a - b).join(', ') : '');
+            }
+            if (facetObject[property]) {
+                facetObject[property] += 1;
+            } else {
+                facetObject[property] = 1;
+            }
+        });
+        // We also want to display terms that could be added if the user wants to increase the displayed results
+        // These terms need to satisfy all of the filters that are already selected
+        newFileList.forEach((file) => {
+            let property = file[propertyKey];
+            if (propertyKey === 'biological_replicates') {
+                property = (file.biological_replicates ? file.biological_replicates.sort((a, b) => a - b).join(', ') : '');
+            }
+            // We only want to look at files that are not in 'filteredListFiltered'
```

Comment:
'fileListFiltered'

Reply:
Looking at it again, I agree `array` is not actually mutated.